### ISSUE TITLE
Make calculation asynchronous and show progress

### ISSUE THREAD

**Issue by sethladd** [swlegion/rollcrits#9]
At least on my pixel 2, pressing buttons has a laggy response. Not sure the cause. Maybe a web worker would help?

**Comment by matanlurey** [comment]
Thanks! I re-wrote the title a bit to I think more properly explain your frustration.

I think web workers could easily be a solution but I'd even just experiment with pausing calculation a bit to give the UI time to breathe or adding an option to reduce the number of rolls to make the calculation faster (especially on mobile):

* [ ] Delay/buffer computation a bit if a user rapidly clicks
* [ ] Run the computation in the background to avoid locking the UI
* [ ] Show a progress bar for computing
* [ ] Show a bar with "results updated" with an option to jump to the results

**Comment by Im-Helping** [comment]
Even on desktop it isn't always clear that the result has actually happened, the results are below the fold at most resolutions (up to 1440 at standard Chrome zoom level); moving the graph to be in line with the attacker die pool would have the results/graph be visible at most times and clear up a lot of unused white space

**Comment by matanlurey** [comment]
So you mean put the graph on top? Or after the attacker?

I could also show a "Results Updated" that if clicked scrolls to the bottom.

**Comment by Im-Helping** [comment]
![image](https://user-images.githubusercontent.com/52722556/60927505-07534000-a25f-11e9-9908-85a455e37bb5.png)

**Comment by matanlurey** [comment]
Got it. I've been trying to be "mobile first" to avoid it looking bad on mobile but that is sensible.

**Comment by Im-Helping** [comment]
Fair point, I think putting it on top on mobile or when resized down makes sense; an interim results updated button is likely 1/100th the work

**Comment by matanlurey** [comment]
> an interim results updated button is likely 1/100th the work

Sounds like my kind of plan. I've updated the request above.
